### ISSUE TITLE
Patch our local version of seaborn to avoid seaborn--numpy warning

### ISSUE THREAD

**Issue by fsaad** [probcomp/iventure#44]
![image](https://cloud.githubusercontent.com/assets/5721897/26362079/f1d67328-3faa-11e7-887a-5ae5a16a4e01.png)

**Comment by riastradh-probcomp** [comment]
Hmm...  Any particular reason we don't just pass a Python list at the caller?

**Comment by riastradh-probcomp** [comment]
I don't understand how this can even happen.  What code path from `.clustermap` can possibly lead to passing anything other than a list as xticklabels/yticklabels?  I see a possible path from `.heatmap`, but not `.clustermap`!

In particular, I'm pretty sure `pivot.columns.tolist()` and `pivot.index.tolist()` both give Python lists, never numpy arrays.

**Comment by fsaad** [comment]
@riastradh-probcomp Will investigate -- it might be an internal seaborn conversion.

**Comment by riastradh-probcomp** [comment]
Do you have a reduced test case handy of a dataframe I can feed into `utils_plot.clustermap` that triggers the warning?

**Comment by fsaad** [comment]
@riastradh-probcomp The image shows a reproducible minimal working example for the warning; I believe utils_plot.clustermap takes a dataframe exactly as shown in `Out[18]` and then does the pivoting internally.

**Comment by riastradh-probcomp** [comment]
No warning raised, goes straight to exception:

```
def test_clustermap_warn():
    df = pd.DataFrame(
        [
            ['f','f',1.0], ['f','b',0.2], ['f','c',0.3],
            ['b','f',0.2], ['b','b',1.0], ['b','c',0.7],
        ],
        columns=('name0', 'name1', 'value'),
    )
    uplt.clustermap(df)
    raise Exception
```

**Comment by fsaad** [comment]
```
probcomp-1:/scratch/fsaad$ ipython
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
Type "copyright", "credits" or "license" for more information.

IPython 5.3.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import iventure.utils_plot as uplt

In [2]: def test_clustermap_warn():
   ...:         df = pd.DataFrame(
   ...:             [
   ...:                 ['f','f',1.0], ['f','b',0.2], ['f','c',0.3],
   ...:                 ['b','f',0.2], ['b','b',1.0], ['b','c',0.7],
   ...:             ],
   ...:             columns=('name0', 'name1', 'value'),
   ...:         )
   ...:         uplt.clustermap(df)
   ...:         raise Exception
   ...: 

In [3]: import pandas as pd

In [4]: test_clustermap_warn()
/scratch/fsaad/.pyenv2.7/local/lib/python2.7/site-packages/iventure/seaborn/matrix.py:143: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  if xticklabels == []:
/scratch/fsaad/.pyenv2.7/local/lib/python2.7/site-packages/iventure/seaborn/matrix.py:151: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  if yticklabels == []:
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-4-b25ef3e20a9a> in <module>()
----> 1 test_clustermap_warn()

<ipython-input-2-86394f766af8> in test_clustermap_warn()
      8         )
      9         uplt.clustermap(df)
---> 10         raise Exception

Exception: 
```

**Comment by fsaad** [comment]
Ah, seems that this warning is based on a local version of seaborn.

**Comment by riastradh-probcomp** [comment]
OK, I see the path inside seaborn that converts it to a numpy array -- inside seaborn.matrix.plot_matrix.  Pooh.

**Comment by riastradh-probcomp** [comment]
Merged in 95dd5b9d93aff5bc4cbd57edd4800a30e35265f0, reported upstream in https://github.com/mwaskom/seaborn/pull/1195.
